go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kueue
 
-go 1.22
+go 1.22.0
```

Comment:
It's updated by `go mod tidy` and is based on the version of the required modules in the current case we have 
```bash
$ go list -m -f "{{.GoVersion}}  {{.Path}}" all | grep "1\.22\."
1.22.0  sigs.k8s.io/kueue
1.22.0  k8s.io/api
1.22.0  k8s.io/apimachinery
1.22.0  k8s.io/kms
```
triggering the change.

The latest release is not relevant since it sets the minimal version needed.